src/nvim/mapping.c

```diff
@@ -2816,20 +2817,28 @@ void modify_keymap(uint64_t channel_id, Buffer buffer, bool is_unmap, String mod
 /// @param  mode  The abbreviation for the mode
 /// @param  buf  The buffer to get the mapping array. NULL for global
 /// @returns Array of maparg()-like dictionaries describing mappings
-ArrayOf(Dictionary) keymap_array(String mode, buf_T *buf)
+ArrayOf(Dictionary) keymap_array(String mode, buf_T *buf, Error *err)
 {
   Array mappings = ARRAY_DICT_INIT;
+  if (mode.size == 1 && mode.data[0] == ' ') {
+    api_set_error(err, kErrorTypeValidation, "Invalid mode name %s", mode.data);
+    return mappings;
+  }
```

Comment:
Hmm, this is even stranger, as it only gives an error for a space but still treats any other invalid mode short-name as nvo-mode. It should either be consistent with the checks in `modify_keymap()` or never give an error.

Reply:
just remove this space check right?